pkg/syncer/metadatasyncer.go

```diff
@@ -288,6 +291,91 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 				log.Infof("Supervisor CNS-CSI FSS: %q is enabled", common.WorkloadDomainIsolationFSS)
 			}
 		}
+
+		parseBool := func(CfgMap *v1.ConfigMap, featureName string, namespace string) bool {
+			var fssVal bool
+			if state, ok := CfgMap.Data[featureName]; ok {
+				fssVal, err = strconv.ParseBool(state)
+				if err != nil {
+					log.Errorf("failed while converting %s FSS with value: %+v "+
+						"to boolean in %q configmap in %q namespace. Killing the container...",
+						featureName, fssVal, CfgMap.Name, namespace)
+					os.Exit(1)
+				}
+			}
+			return fssVal
+		}
+		if !IsWorkloadDomainIsolationSupported {
+			// If the WCP capability for WorkloadDomainIsolation feature is disabled,
+			// start an informer to know when the capability is enabled and restart the container.
+			err = metadataSyncer.k8sInformerManager.AddConfigMapListener(
+				ctx,
+				k8sClient,
+				common.KubeSystemNamespace,
+				// Add.
+				nil,
+				// Update.
+				func(oldObj interface{}, newObj interface{}) {
```

Comment:
We will not be able to add another update call back. Only one update call back can be added.

Refer to

```golang
// OnUpdate calls UpdateFunc if it's not nil.
func (r ResourceEventHandlerFuncs) OnUpdate(oldObj, newObj interface{}) {
	if r.UpdateFunc != nil {
		r.UpdateFunc(oldObj, newObj)
	}
}
```

Reply:
Using a new configmaplistener which is not part of the informer factory